tests/validator_churn.go

```diff
@@ -0,0 +1,158 @@
+		teleporterContractAddress, subnetAInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+	subnetBTeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress, subnetBInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+
+	ctx := context.Background()
+
+	//
+	// Send a Teleporter message on Subnet A
+	//
+	log.Info("Sending Teleporter message on source chain", "destinationChainID", subnetBInfo.BlockchainID)
+	sendCrossChainMessageInput := teleportermessenger.TeleporterMessageInput{
+		DestinationChainID: subnetBInfo.BlockchainID,
+		DestinationAddress: fundedAddress,
+		FeeInfo: teleportermessenger.TeleporterFeeInfo{
+			ContractAddress: fundedAddress,
+			Amount:          big.NewInt(0),
+		},
+		RequiredGasLimit:        big.NewInt(1),
+		AllowedRelayerAddresses: []common.Address{},
+		Message:                 []byte{1, 2, 3, 4},
+	}
+
+	var receipt *types.Receipt
+	receipt, teleporterMessageID = utils.SendCrossChainMessageAndWaitForAcceptance(
+		ctx,
+		subnetAInfo,
+		subnetBInfo,
+		sendCrossChainMessageInput,
+		fundedAddress,
+		fundedKey,
+		subnetATeleporterMessenger,
+	)
+
+	sendEvent, err := utils.GetEventFromLogs(receipt.Logs, subnetATeleporterMessenger.ParseSendCrossChainMessage)
+	Expect(err).Should(BeNil())
+	sentTeleporterMessage := sendEvent.Message
+
+	// Construct the signed warp message
+	signedWarpMessageBytes := localUtils.ConstructSignedWarpMessageBytes(ctx, receipt, subnetAInfo, subnetBInfo)
+
+	//
+	// Modify the validator set on Subnet A
+	//
+
+	// Add new nodes to the validator set
+	log.Info("Adding nodes to the validator set")
+	var nodesToAdd []string
+	for i := 15; i <= 19; i++ {
+		n := fmt.Sprintf("node%d-bls", i)
+		nodesToAdd = append(nodesToAdd, n)
+	}
+	localUtils.AddSubnetValidators(ctx, subnetAInfo.SubnetID, nodesToAdd)
+
+	// Refresh the subnet info
+	subnets = network.GetSubnetsInfo()
+	subnetAInfo = subnets[0]
+	subnetBInfo = subnets[1]
+
+	// Trigger the proposer VM to update its height so that the inner VM can see the new validator set
+	err = subnetEvmUtils.IssueTxsToActivateProposerVMFork(
+		ctx, subnetAInfo.ChainIDInt, fundedKey, subnetAInfo.ChainWSClient,
+	)
+	Expect(err).Should(BeNil())
+	err = subnetEvmUtils.IssueTxsToActivateProposerVMFork(
+		ctx, subnetBInfo.ChainIDInt, fundedKey, subnetBInfo.ChainWSClient,
+	)
+	Expect(err).Should(BeNil())
+
+	//
+	// Attempt to deliver the warp message signed by the old validator set. This should fail.
+	//
+	// Construct the transaction to send the Warp message to the destination chain
+	signedTx := utils.CreateReceiveCrossChainMessageTransaction(
+		ctx,
+		signedWarpMessageBytes,
+		sendEvent.Message.RequiredGasLimit,
+		teleporterContractAddress,
+		fundedAddress,
+		fundedKey,
+		subnetBInfo,
+		false,
+	)
+
+	log.Info("Sending transaction to destination chain")
+	receipt = utils.SendTransactionAndWaitForAcceptance(ctx, subnetBInfo, signedTx, false)
```

Comment:
I think we can just use `network.RelayMessage` here and expect failure

Reply:
Here, we want to deliver the message signed by the old validator set. If we use `network.RelayMessage`, it will construct a new aggregate signature using the updated validator set.